sw/device/tests/sram_ctrl_smoketest.c

```diff
@@ -0,0 +1,89 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_sram_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+////// TEST MAIN
+
+bool test_main() {
+  LOG_INFO("Running SRAM_CTRL DIF test...");
+
+  // Initialize SRAM_CTRL hardware.
+  dif_sram_ctrl_t sram_ctrl_main;
+  dif_sram_ctrl_t sram_ctrl_ret;
+  CHECK_DIF_OK(
+      dif_sram_ctrl_init(mmio_region_from_addr(TOP_EARLGREY_SRAM_CTRL_MAIN_REGS_BASE_ADDR), &sram_ctrl_main));
+  CHECK_DIF_OK(
+      dif_sram_ctrl_init(mmio_region_from_addr(TOP_EARLGREY_SRAM_CTRL_RET_AON_REGS_BASE_ADDR ), &sram_ctrl_ret));
+
+  dif_sram_ctrl_status_bitfield_t status_main;
+  dif_sram_ctrl_status_bitfield_t status_ret;
+
+  // Check Status registers
+  CHECK_DIF_OK(dif_sram_ctrl_get_status(&sram_ctrl_ret, &status_ret));
+  CHECK_DIF_OK(dif_sram_ctrl_get_status(&sram_ctrl_main, &status_main));
+
+  // check return value for main sram status, key has been requested in bootrom
+
+  CHECK(status_main & (kDifSramCtrlStatusScrKeyValid |
+                       kDifSramCtrlStatusScrKeySeedValid |
+                       kDifSramCtrlStatusInitDone),
+        "SRAM_CTRL DIF test SRAM MAIN Init Expected Value Mismatch");
+
+  // check return value for ret sram status, key has not been requested, expect 0x0
+
+  CHECK(status_ret == 0x0,"SRAM_CTRL DIF test SRAM RET Init Expected Value Mismatch");
+
+  LOG_INFO("SRAM_CTRL DIF test status registers checked");
+
+  // Read and Write to/from SRAMs
+
+  mmio_region_t sram_region_main;
+  mmio_region_t sram_region_ret;
+  uint32_t      rw_data_32;
+
+  sram_region_main = mmio_region_from_addr(TOP_EARLGREY_SRAM_CTRL_MAIN_RAM_BASE_ADDR);
+  sram_region_ret  = mmio_region_from_addr(TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR);
+
+  // ret SRAM first
+  // small number of writes/reads for basic smoketest
+  for (int i=0; i<16; i++) {
+    rw_data_32 = (i<<26|i<<19|i<<11|i<<3|i>>5);
+    mmio_region_write32(sram_region_ret, i<<2, rw_data_32);
+  }
+  for (int i=0; i<16; i++) {
+    rw_data_32 = mmio_region_read32(sram_region_ret, i<<2);
+    CHECK(rw_data_32 == (i<<26|i<<19|i<<11|i<<3|i>>5),"Memory Write/Read Mismatch for SRAM RET");
+  }
+
+  LOG_INFO("SRAM_CTRL DIF test SRAM RET Write/Read Done");
+
+  // main SRAM next
+  // small number of writes/reads for basic smoketest
+  for (int i=0; i<16; i++) {
+    rw_data_32 = (i<<26|i<<19|i<<11|i<<3|i>>5);
```

Comment:
nit: one space around `=`, `<` and `<<`, similar for other lines
```suggestion
  for (int i = 0; i < 16; i++) {
    rw_data_32 = (i << 26 | i << 19 | i << 11 | i << 3 | i >> 5);
```

Reply:
more style guidelines can be found here https://github.com/lowRISC/style-guides/blob/master/VerilogCodingStyle.md